pkg/reconciler/workload/deploymentsplitter/deploymentsplitter_controller.go

```diff
@@ -50,7 +53,8 @@ func NewController(cfg *rest.Config) *Controller {
 	client := appsv1client.NewForConfigOrDie(cfg)
 	kubeClient := kubernetes.NewForConfigOrDie(cfg)
 	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "kcp-deployment")
-	stopCh := make(chan struct{}) // TODO: hook this up to SIGTERM/SIGINT
+	stop := context.TODO()
+	stop, _ = signal.NotifyContext(stop, os.Interrupt)
```

Comment:
This seemed like a reasonable thing to do for now, we needed a context anyway to plumb the loggers

Reply:
this whole code here will be rewritten soonish